contributors/design-proposals/federated-resource-quota.md

```diff
@@ -0,0 +1,53 @@
+# Federated ResourceQuota Requirements & Design
+@weijinxu, @IrfanUrRehman, @JianhuiZ
+reviewed by  @quinton-hoole, @deepak-vijSummary
+
+In a single Kubernetes cluster there is a concept of Resource Quota that manages the resources by namespace so that user usages can be limited to their fair share of the resources. We want to expand this concept to the federation level so that users can treat Federation Control Plane as a “Single Pane of Glass” for seamlessly managing the Resource Quota  information across all the underlying federated clusters.
+
+## User scenario
+
+As a federation user and an admin of clusters, I want to be able to define an overall resource quota for some specific or all resources for a given federated namespace. The overall resource quota limit should be applicable across clusters and preferably enforced at federation control plane level.
+
+For example, let us say that we’ve specified 100 pods in the resource quota spec in the federation namespace, with multiple underlying federated clusters. Let us assume that 50 pods have been deployed already across the federation in a previous deployment run. In a subsequent deployment run, if a user requests additional 50+ more pods to be deployed, the deployment will be rejected as a whole by the federation control plane before deployment process actually starts. If user requests less than 50 pods, the deployment should succeed as per the deployment preferences.
```

Comment:
quota today is not enforced on the controller replica count, but on the literal pod creation request.

this text implies to me that we will try to enforce quota about pods with knowledge about controller replica count.  i don't think we should do that at this time.  i think have general object count quota for federated api types is fine, but since pods are not in the control plane itself, this seemed problematic.